### PR TITLE
Feature - 6021 - screening questions backend setup

### LINKED ISSUE #6021: Backend: add Screening Questions to Pools
## ✨ Feature

Add screening questions added to Pool. They get their own table, with a one-to-many relationship with Pools.

## 🕵️ Details

- Screening Questions table added to database
- Relationship added to Pool model
- Pool factory automatically adds a couple screening questions
- ScreeningQuestion type and its relationship to PoolAdvertisement added to schema
- Screening Questions can be added, updated and removed in updatePoolAdvertisement mutation
- Permissions are the same as the pool

## 🙋‍♀️ Proposed Implementation

Suggested model changes:
```
ScreeningQuestion {
  id: ID!
  question: LocalizedString!
  sortOrder: number
  poolAdvertisment: PoolAdvertisement
}

PoolAdvertisement {
  ...
  screeningQuestions: [ScreeningQuestion]
}
```

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] When querying a PoolAdvertisment, can get a list of associated Screening Questions, which have a text field and an id
- [ ] Pool factory automatically adds a couple screening questions to generated pools
- [ ] When updating a PoolAdvertisment with a mutation, Screening Questions can be edited, added or removed.
- [ ] PHPUnit test(s) for the updated mutation (and query?)

## :stop_sign: Blockers

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #6021

## 👋 Introduction

Adds screening questions to the database and API backend 

## 🕵️ Details

Migration, model files, factory, and schema changes

## 🧪 Testing

1. try seeding and querying pools
2. try creating/updating/deleting questions

sample query to add a question to a pool
you need a draft pool

```
mutation createQuestion {
  updatePoolAdvertisement(
    	id: "", 
    	poolAdvertisement: {
    		screeningQuestions: {
    			create: [
          	{
      				question: {
        				en: "aaa",
        				fr: "bbb",
      				}
          	},
        	]
    	        }
  	}
)
  {
    screeningQuestions {
      question {
        en
        fr
      }
    }
  }
}
```

### CODE REVIEW

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -1568,6 +1577,23 @@ input CreatePoolAdvertisementInput {
...
 }
 
+input CreateScreeningQuestionInput {
+  question: LocalizedStringInput!
+  sortOrder: Int @rename(attribute: "sort_order")
+}
+
+input UpdateScreeningQuestionInput {
+  id: ID!
+  question: LocalizedStringInput!
```
Comment (petertgiles) [inline api/graphql/schema.graphql:1587]: For updates, this should be nullable in case you don't want to change the questions.

**Review by petertgiles — CHANGES_REQUESTED** [review]
Looking solid!  :clap:  A few things to tidy up...

File: api/database/factories/PoolFactory.php [hunk 2]
```diff
@@ -64,6 +65,13 @@ public function configure()
                 $pool->stream = $this->faker->randomElement(ApiEnums::poolStreams());
                 $pool->save();
             }
+
+            for ($i = 1; $i < 4; $i++) {
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/database/factories/PoolFactory.php]: sure, done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6089/commits/d1480e1a3c6f781be8b37abc94219434a912503f

File: api/graphql/schema.graphql (hunk 1 above)
Comment (vd1992) [reply api/graphql/schema.graphql]: yeah, that is true https://github.com/GCTC-NTGC/gc-digital-talent/pull/6089/commits/e510bc5c24416f555f6a709f1b1a093c249e7818

File: api/graphql/schema.graphql [hunk 3]
```diff
@@ -1568,6 +1577,23 @@ input CreatePoolAdvertisementInput {
...
+  sortOrder: Int @rename(attribute: "sort_order")
+}
+
+input UpdateScreeningQuestionsHasMany {
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql:1593]: Does disconnect do the same thing as delete? The pool_id is non-nullable for screening question so the question has to be deleted

[the remark this answers is not in the document]
Comment (petertgiles) [reply api/graphql/schema.graphql:1593]: Oh, you're right, I'm confusing it with BelongsToMany.  Yep, follow to the docs, for sure.

**Review by petertgiles — APPROVED** [review]

**Pull request merged**